src/app/compressed-air-assessment/compressed-air-assessment-results.service.ts

```diff
@@ -1056,6 +1057,76 @@ export class CompressedAirAssessmentResultsService {
     });
   }
 
+  getProfileSummaryDataAverages(profileSummary: Array<ProfileSummary>): Array<ProfileSummary>{
+    let updatedSum: Array<ProfileSummary> = new Array<ProfileSummary>();
+    profileSummary.forEach(sums => {
+      let updatedData: ProfileSummary = sums;
+      updatedData.avgPower = this.getAvgPower(sums.profileSummaryData);
```

Comment:
You can use _meanBy from the lodash library for these averages. Used a few other times in the project if you do a ctrl + f. Per Mark... `updatedData.avgPower = _.meanBy(sums.profileSummaryData, 'power')`

Reply:
`updatedData.avgPower = _.meanBy(sums.profileSummaryData, 'power')` So this solution for calculating the averages dose not work because it does not take into account "OFF" and "ON" days